finetune/adapter.py

```diff
@@ -218,7 +218,8 @@ def validate(
     for k in range(eval_iters):
         input_ids, targets = get_batch(fabric, val_data, longest_seq_length)
         logits = model(input_ids)
-        loss = chunked_cross_entropy(logits, targets, chunk_size=0)
+        logits[-1] = logits[-1][..., :-1, :]
```

Comment:
You don't need this list modification because `model.forward(lm_head_chunk_size=0)` by default. Instead, copy what's done for `full.py`

All the chunk size tricks are only relevant for `backward`, so you don't need them at validation

Reply:
sorry not sure i follow?

Reply:
you mean just use `logits[..., :-1, :], targets[..., 1:]`?

Reply:
Logits is only a list for the finetuning models when `lm_head_chunk_size>0` (not the default)

https://github.com/Lightning-AI/lit-gpt/blob/main/lit_gpt/lora.py#L532-L535

Which means that you want to do

```python
        chunked_cross_entropy(logits[..., :-1, :], targets[..., 1:], chunk_size=0)
```

as with `full.py`